File: selection/_012/Main.java
package selection._012;

import java.util.ArrayList;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        int[] x = new int[m];
        int[] y = new int[m];
        for (int i = 0; i < m; i++) {
            x[i] = sc.nextInt();
            y[i] = sc.nextInt();
        }
        sc.close();

        int result = 0;
        for (int i = 1; i < 1<<n; i++) {
            ArrayList<Integer> candidates = new ArrayList<>();
            for (int j = 0; j < n; j++) {
                if ((i>>j)%2==1) {
                    candidates.add(j + 1);
                }
            }
            int size = candidates.size();
            boolean ok = true;
            for (int j = 0; j < size; j++) {
                for (int j2 = j + 1; j2 < size; j2++) {
                    boolean check = false;
                    for (int k = 0; k < m; k++) {
                        if ((x[k] == candidates.get(j) && y[k] == candidates.get(j2)) || (x[k] == candidates.get(j2) && y[k] == candidates.get(j))) {
                            check = true;
                        }
                    }
                    if (!check) {
                        ok = false;
                    }
                }
            }
            if (ok) {
                result = Math.max(result, size);
            }
        }

        System.out.println(result);
    }
}

File: selection/_012/main.go
package main

import (
	"fmt"
	"math"
)

type edge struct {
	x, y int
}

func main() {
	var n, m int
	fmt.Scan(&n, &m)

	x := make([]int, m)
	y := make([]int, m)
	g := make(map[edge]bool)
	for i := 0; i < m; i++ {
		fmt.Scan(&x[i], &y[i])
		g[edge{x: x[i] - 1, y: y[i] - 1}] = true
		g[edge{x: y[i] - 1, y: x[i] - 1}] = true
	}

	result := 0
	for i := 0; i < 1<<n; i++ {
		a := make([]int, 0)
		for j := 0; j < n; j++ {
			if (i & (1 << j)) > 0 {
				a = append(a, j)
			}
		}

		ok := true
		for _, v := range a {
			for _, vv := range a {
				if v == vv {
					continue
				}

				if !g[edge{v, vv}] {
					ok = false
				}
			}
		}

		if ok {
			result = int(math.Max(float64(result), float64(len(a))))
		}
	}

	fmt.Println(result)
}
